embassy-stm32/src/usart/mod.rs

```diff
@@ -1051,7 +1066,21 @@ impl<'d> Uart<'d, Async> {
         Self::new_inner(
             peri,
             None,
-            new_pin!(tx, AfType::output(OutputType::PushPull, Speed::Medium)),
+            new_pin!(
+                tx,
+                match half_duplex {
```

Comment:
this is copypasted 4 times, can you extract it to a method like `HalfDuplexConfig::af_type()`?